warehouse/integrations/deltalake-native/deltalake_native.go

```diff
@@ -0,0 +1,1356 @@
+package deltalake_native
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	dbsqllog "github.com/databricks/databricks-sql-go/logger"
+
+	dbsql "github.com/databricks/databricks-sql-go"
+	"github.com/rudderlabs/rudder-server/warehouse/logfield"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	warehouseclient "github.com/rudderlabs/rudder-server/warehouse/client"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+const (
+	Host                   = "host"
+	Port                   = "port"
+	Path                   = "path"
+	Token                  = "token"
+	Catalog                = "catalog"
+	UseSTSTokens           = "useSTSTokens"
+	EnableExternalLocation = "enableExternalLocation"
+	ExternalLocation       = "externalLocation"
+	UserAgent              = "Rudderstack"
+)
+
+const (
+	provider       = warehouseutils.DELTALAKE
+	tableNameLimit = 127
+)
+
+const (
+	schemaNotFound       = "[SCHEMA_NOT_FOUND]"
+	partitionNotFound    = "SHOW PARTITIONS is not allowed on a table that is not partitioned"
+	columnsAlreadyExists = "already exists in root"
+)
+
+const (
+	mergeMode  = "MERGE"
+	appendMode = "APPEND"
+)
+
+var pkgLogger logger.Logger
+
+// dataTypesMap maps rudder data types to delta lake data types
+var dataTypesMap = map[string]string{
+	"boolean":  "BOOLEAN",
+	"int":      "BIGINT",
+	"float":    "DOUBLE",
+	"string":   "STRING",
+	"datetime": "TIMESTAMP",
+	"date":     "DATE",
+}
+
+// dataTypesMapToRudder maps delta lake data types to rudder data types
+// Reference: https://docs.databricks.com/sql/language-manual/sql-ref-datatype-rules.html
+var dataTypesMapToRudder = map[string]string{
+	"TINYINT":   "int",
+	"SMALLINT":  "int",
+	"INT":       "int",
+	"BIGINT":    "int",
+	"DECIMAL":   "float",
+	"FLOAT":     "float",
+	"DOUBLE":    "float",
+	"BOOLEAN":   "boolean",
+	"STRING":    "string",
+	"DATE":      "date",
+	"TIMESTAMP": "datetime",
+	"tinyint":   "int",
+	"smallint":  "int",
+	"int":       "int",
+	"bigint":    "int",
+	"decimal":   "float",
+	"float":     "float",
+	"double":    "float",
+	"boolean":   "boolean",
+	"string":    "string",
+	"date":      "date",
+	"timestamp": "datetime",
+}
+
+// excludeColumnsMap Columns you need to exclude
+// Since event_date is an auto generated column in order to support partitioning.
+// We need to ignore it during query generation.
+var excludeColumnsMap = map[string]bool{
+	"event_date": true,
+}
```

Comment:
[minor optimization] we don't really need a boolean here. using a `struct{}` instead should have a [smaller footprint](https://stackoverflow.com/a/22770744/828366):

```suggestion
var excludeColumnsMap = map[string]struct{}{
	"event_date": {},
}
```